cli/azd/pkg/ext/hooks_manager.go

```diff
@@ -142,17 +143,51 @@ func (h *HooksManager) ValidateHooks(ctx context.Context, allHooks map[string][]
 	}
 
 	hasPowerShellHooks := false
+	hasDefaultShellHooks := false
 
-	// Check all hooks
+	// First, perform lightweight validation to set flags like usingDefaultShell
+	// without creating temporary files (which full validation does)
+	for _, hookConfigs := range allHooks {
+		for _, hookConfig := range hookConfigs {
+			// Set the working directory for validation
+			if hookConfig.cwd == "" {
+				hookConfig.cwd = h.cwd
+			}
+
+			// Only perform shell detection for warning purposes, not full validation
+			if !hookConfig.validated && hookConfig.Run != "" {
+				// Check if it's an inline script (no file exists)
+				relativeCheckPath := strings.ReplaceAll(hookConfig.Run, "/", string(os.PathSeparator))
+				fullCheckPath := relativeCheckPath
+				if hookConfig.cwd != "" {
+					fullCheckPath = filepath.Join(hookConfig.cwd, hookConfig.Run)
+				}
+
+				_, err := os.Stat(fullCheckPath)
+				isInlineScript := err != nil // File doesn't exist, so it's inline
+
+				// If shell is not specified and it's an inline script, set default for warning
+				if hookConfig.Shell == ScriptTypeUnknown && isInlineScript {
+					if runtime.GOOS == "windows" {
+						hookConfig.Shell = ShellTypePowershell
+					} else {
+						hookConfig.Shell = ShellTypeBash
+					}
+					hookConfig.usingDefaultShell = true
+				}
+			}
+		}
+	}
+
+	// Now check all hooks for warning conditions
```

Comment:
We're doing 2 full passes on all the hooks now - could we just perform this same validation in the same pass?

Reply:
yeah, I'll add a note about why we need a 2-step validation.
Hopefully not that bad b/c the number of hooks it likely never that big to hit performance